runtime/src/main/cpp/ObjCExport.mm

```diff
@@ -264,11 +261,21 @@ - (instancetype)copyWithZone:(NSZone *)zone {
         freeWhenDone:NO] autorelease];
   } else {
     // TODO: consider making NSString subclass to avoid copying here.
-    NSString* result = [[NSString alloc] initWithBytes:utf16Chars
+    NSString* candidate = [[NSString alloc] initWithBytes:utf16Chars
```

Comment:
Unrelated to this change, but https://developer.apple.com/documentation/foundation/nsstring/1413830-initwithbytesnocopy?language=objc may be better choice here, lifetime of Kotlin and Obj-C objects likely match, so we can use single storage.

Reply:
> lifetime of Kotlin and Obj-C objects likely match,

Unfortunately, this is not correct: default `NSString` implementation can't prevent Kotlin object from removing. Consider the case like
```kotlin
// Kotlin
fun toString(x: Int) = x.toString()
```

```swift
// Swift
let str = TestKt.toString(x: 42)
// Kotlin string doesn't have any references to it anymore.
```

 We need custom `NSString` implementation holding Kotlin object reference like `KotlinBase` does. `TODO` above refers exactly to this approach.